src/App.tsx

```diff
@@ -1,51 +1,88 @@
 
-// import goodsFromServer from './goods';
+type State = {
+  selectedGood: string;
+};
 
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: '',
+  };
 
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
+  render() {
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
 
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
+            {this.state.selectedGood
+              ? `${this.state.selectedGood} is selected`
+              : 'No goods selected'}
+          </h1>
 
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
+          <button
+            type="button"
+            className={this.state.selectedGood === ''
+              ? 'App__clear'
+              : 'App__clear--visible'}
+            onClick={() => {
+              this.setState({ selectedGood: '' });
+            }}
+          >
+            Clear
+          </button>
+        </header>
 
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
-};
+        <ul>
+          {goodsFromServer.map(good => (
+            <li className={
+              this.state.selectedGood.includes(good)
+                ? 'Good--active'
+                : 'Good'
+            }
+            >
+              {good}
+              <button
+                type="button"
+                className={
+                  this.state.selectedGood.includes(good)
+                    ? 'Good__select'
+                    : 'Good__select--visible'
+                }
+                onClick={() => {
+                  this.setState({
+                    selectedGood: good,
+                  });
+                }}
+              >
+                Select
+              </button>
+
+              <button
+                type="button"
+                className={
+                  this.state.selectedGood.includes(good)
+                    ? 'Good__remove--visible'
+                    : 'Good__remove'
+                }
+                onClick={() => {
+                  this.setState({ selectedGood: '' });
+                }}
```

Comment:
you've repeated the logic that you have on `Clear` button, you should definitely move this func to a handler